common/djangoapps/third_party_auth/tests/test_pipeline.py

```diff
@@ -75,7 +75,7 @@ def setUp(self):
         ('S', 'S9fe2', False),
         ('S', 'S9fe2', True),
         ('S.K', 'S_K', False),
-        ('S.K.', 'S_K_', False),
+        ('S.K.', 'S_K', False),
         ('S.K.', 'S_K_9fe2', True),
```

Comment:
so with the change of logic wouldn't line 79's value just be `S_K9fe2`

Reply:
@taliaedX Since we are joining the substrings returned by regex find all back again with an underscore`_`, it will only remove the underscore at the end of the username if there is any. The underscore just before `_9fe2` is included by the pipeline's `get_username()` function.

Reply:
Ahhhh I didn't realize it's appending the UUID _before_ calling the clean_username function, thus the second underscore is no longer at the end of the username.

I was thinking the other way around and thus the join logic wouldn't append the second underscore.